instrumentation/spring/spring-boot-autoconfigure/src/main/java/io/opentelemetry/instrumentation/spring/autoconfigure/internal/properties/SpringConfigProperties.java

```diff
@@ -29,37 +31,92 @@ public class SpringConfigProperties implements ConfigProperties {
   private final ExpressionParser parser;
   private final OtlpExporterProperties otlpExporterProperties;
   private final OtelResourceProperties resourceProperties;
-  private final PropagationProperties propagationProperties;
+  private final OtelSpringProperties otelSpringProperties;
   private final ConfigProperties otelSdkProperties;
+  private final ConfigProperties customizedListProperties;
+
+  private static final Map<String, Function<OtelSpringProperties, List<String>>> LIST_PROPERTIES =
```

Comment:
I don't see how this can work, e.g. here - but maybe you have an idea

```   
LIST_PROPERTIES.put(
        "otel.experimental.resource.disabled.keys",
        OtelSpringProperties::getExperimentalResourceDisabledKeys);
```

Reply:
I may miss something. Something like that?

```java

 private static final Map<String, String> VALUE_FOR_LIST_PROPERTY = new HashMap<>();


  public SpringConfigProperties(
      Environment environment,
      ExpressionParser parser,
      OtlpExporterProperties otlpExporterProperties,
      OtelResourceProperties resourceProperties,
      OtelSpringProperties otelSpringProperties,
      ConfigProperties otelSdkProperties) {
     
  ...
   VALUE_FOR_LIST_PROPERTY.put(ENABLED_KEY, otelSpringProperties.getJavaEnabledResourceProviders()):
 ...
)


  @SuppressWarnings("unchecked")
  @Override
  public List<String> getList(String name) {

    String normalizedName = ConfigUtil.normalizeEnvironmentVariableKey(name);

    List<String> propListValues = VALUE_FOR_LIST_PROPERTY.get(normalizedName)
   
    if(propListValues != null &&  !propListValues.isEmpty()) {
      List<String> c = customizedListProperties.getList(normalizedName);
      if (!c.isEmpty()) {
        return c;
      }
      return VALUE_FOR_LIST_PROPERTY.get(normalizedName);
    }

    return or(environment.getProperty(normalizedName, List.class), otelSdkProperties.getList(name));
  }

```

Reply:
great idea - thanks :smile: